capstone/src/main/webapp/flags_script.js

```diff
@@ -10,25 +10,30 @@
 // distributed under the License is distributed on an 'AS IS' BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License.
-//global vars; referenced in both this script and locations_script 
+// limitations under the License. 
 let map;
 let heatmap;
 let map_style;
 let establishment_markers = [];
-let heatmap_data = [];
+let heatmap_data_users = [];
 let marker_dict = {};
 let flag_dict = {};
 let user_lat;
 let user_lng;
-let DISTANCE_THRESHOLD_MILES = 15; //predetermined constant; max distance to be considered "close" to user
+let sorted_report_counts;
+let total_reports_count = 0;
+let report_counts_dict = {};
+ 
+//constants 
+let DISTANCE_THRESHOLD_MILES = 15; //max distance to be considered "close" to user
 let EARTH_RADIUS_MILES = 3958.8;
+let SWITCH_HOUR = 18; //6:00 pm
```

Comment:
It looks like you are modifying a lot of global state in this file. Can you avoid global state throughout your JavaScript code?